x/staking/legacy/v040/migrate.go

```diff
@@ -0,0 +1,116 @@
+package v040
```

Comment:
I'll pay a beer to anyone who tells me if there's a way to make this boilerplate code shorter (esp the 5 for loops and some inner loops...)

Reply:
You may be able to just amino JSON encode and decode from one version to the next. If there were no other changes, the amino JSON should be the same.